charts/ingresslib/templates/_backendconfig.yaml

```diff
@@ -1,4 +1,5 @@
 {{- define "libchart.backendconfig.tpl" -}}
+{{ if .Values.ingress.enabled -}}
```

Comment:
I left the same comment on the other PR-if ingress is disabled, I don't think we want to even `define` anything here, so I don't think the define should come before the `if` (we don't want an unenabled chart to still have side effects, like adding a bunch of empty variables)

Reply:
I don't think so. Because [this line](https://github.com/broadinstitute/terra-helm/blob/48f9ad51b99931f0875189d9561a2f4330390b1c/charts/ingresslib/templates/_backendconfig.yaml#L27) expects ```libchart.backendconfig.tpl``` template to be defined. If I place the if statement before the define, the function may not exist which would cause a template not defined error. An empty template is okay here. Because ```libchart.util.merge```  takes common values between the templates and overrides the values seen here. Also, it adds any additional information.

Reply:
I guess what I'm saying is that literally the entire file should be wrapped in these "if ingress enabled" statements. If the chart isn't enabled, everything here should do absolutely nothing. No references, nothing gets defined.

I agree that an empty template technically works here, but right now "disabled" means "well it still does things but not anything useful" instead of "it does nothing"

Reply:
I don't think you can place if statements outside of a template definition. I tried rendering the chart when the template definition is  wrapped in the if statement, as well as putting the entire file in an if statement and I kept getting the error ```unexpected <define> in command```. I guess this kinda makes sense. In general, I would try not to wrap a function/method/class in an if statement. I wouldn't do something like:

```
if blah:
   def my_function():
      ...
```
or 
```
if blah:
   class ClassDef
```

Reply:
Ah I gotcha, I just looked at what I'd done in the part and you're totally correct, sorry for sending you on a wild goose chase. This looks great, and thank you for the explanation!